### PR TITLE
Add CHANGELOG guidance to link fixed issues

### LINKED ISSUE #988: Update CHANGELOG guidance to reference issues when appropriate
@heaths commented on [Thu Jan 30 2020](https://github.com/Azure/azure-sdk-for-net/issues/9723)

Talked with Adrian offline and he agrees this is acceptable. I will open an issue to update the guidance.

_Originally posted by @heaths in https://github.com/_render_node/MDIzOlB1bGxSZXF1ZXN0UmV2aWV3VGhyZWFkMjMwMTA2NTEzOnYy/pull_request_review_threads/discussion_

### PR DESCRIPTION
**Pull request by heaths**
Resolves #988

### CODE REVIEW

File: docs/policies/releases.md [hunk 1]
```diff
@@ -30,9 +30,13 @@ We use GitHub releases as a convenient place to put release notes. The change lo
 
 ## ChangeLog Guidance
 
-Every package MUST maintain a changelog. Ensuring that a `CHANGELOG.md` file is both available and formatted appropriately will result in automatically formatted release notes on each GitHub release.
+Ensuring that a `CHANGELOG.md` file is both available and formatted appropriately will result in automatically formatted release notes on each GitHub release.
```
Comment (chidozieononiwu) [inline docs/policies/releases.md]: `Every package MUST maintain a changelog` is actually language requested by Arch board members. We might want to leave it.

Comment (heaths) [reply docs/policies/releases.md]: That's captured below using the same MUST/SHOULD/etc. we use elsewhere. @adrianhall?

Comment (heaths) [reply docs/policies/releases.md]: IMO, the images make requirements stand out much better than simple text in a sea of lots of text. 🙂

File: docs/policies/releases.md [hunk 2]
```diff
@@ -62,7 +66,7 @@ Example Changelog
 
 ### Fixed
 - Issue where AccountName on BlobUriBuilder would not be populated
-  for non-IP style Uris.
+  for non-IP style Uris ([#123](https://github.com/Azure/azure-sdk-for-net#123))
```
Comment (weshaggard) [inline docs/policies/releases.md]: Did you want your link to be like `https://github.com/Azure/azure-sdk-for-net/issues/12`

Comment (heaths) [reply docs/policies/releases.md]: Yes, sorry. I was mixing the real link and the shortcut on-site syntax.

File: docs/policies/releases.md (hunk 1 above)
Comment (heaths) [reply docs/policies/releases.md]: Good callout. As discussed offline, this is fine. I figured that one sentence really described *why* we had the (now) following requirements so moved it as more of an explanation.

**Pull request merged**